### PR TITLE
fix ethereumGetAddress and showDisplay param use

### PR DESCRIPTION
**Pull request by natven**
- fix ethereumGetAddress
- fix use of showDisplay param in getAddress and ethereumGetAddress

### CODE REVIEW

File: src/keepkey.ts [hunk 1]
```diff
@@ -172,11 +172,21 @@ export default class KeepKey {
...
-    return addressStr || ('0x' + (address as string))
+    const ethAddress = response as Messages.EthereumAddress
+
+    let result
+    if (ethAddress.hasAddressStr()) {
+      result = ethAddress.getAddressStr()
+    } else if (ethAddress.hasAddress()) {
+      result = '0x' + toHexString(ethAddress.getAddress_asU8())
+    } else {
+      console.log(`unable to obtain eth address from device`)
```
Comment (spiceboi) [inline src/keepkey.ts]: Should we `throw` here if this is an unexpected error?

Comment (keepkeyjon) [reply src/keepkey.ts]: yeah, that'd be prudent

Comment (natven) [reply src/keepkey.ts]: maybe, wasn't sure the best approach. throw instead?

Comment (keepkeyjon) [reply src/keepkey.ts]: throw + early return

Comment (natven) [reply src/keepkey.ts]: prefer multiple return statements? ok

Comment (keepkeyjon) [reply src/keepkey.ts]: yeah. control flow is easier to read than dataflow

Comment (keepkeyjon) [reply src/keepkey.ts]: s/read/reason about/

Comment (natven) [reply src/keepkey.ts]: /g

**Comment by natven** [comment]
modified to throw and early return

**Pull request merged**